src/algebra/jordan/basic.lean

```diff
@@ -0,0 +1,218 @@
+{ lmul_comm_rmul := λ a b, by rw [is_comm_jordan.mul_comm, is_comm_jordan.mul_comm a b],
+  lmul_lmul_comm_lmul := λ a b, by rw [is_comm_jordan.mul_comm (a * a) (a * b),
+  is_comm_jordan.jordan, is_comm_jordan.mul_comm b (a * a)],
+  lmul_comm_rmul_rmul := λ a b, by rw [is_comm_jordan.mul_comm, ←is_comm_jordan.jordan,
+    is_comm_jordan.mul_comm],
+  lmul_lmul_comm_rmul :=  λ a b, by rw [is_comm_jordan.mul_comm (a * a) (b * a),
+    is_comm_jordan.mul_comm b a, is_comm_jordan.jordan, is_comm_jordan.mul_comm,
+    is_comm_jordan.mul_comm b (a * a)],
+  rmul_comm_rmul_rmul := λ a b, by rw [is_comm_jordan.mul_comm b a, is_comm_jordan.jordan,
+    is_comm_jordan.mul_comm], }
+
+universe u
+
+/-- Semigroup multiplication satisfies the (non-commutative) Jordan axioms-/
+@[priority 100] -- see Note [lower instance priority]
+instance semigroup.is_jordan (B : Type u) [semigroup B] : is_jordan B :=
+{ lmul_comm_rmul := λ a b, by rw mul_assoc,
+  lmul_lmul_comm_lmul := λ a b, by rw [mul_assoc, mul_assoc],
+  lmul_comm_rmul_rmul := λ a b, by rw [mul_assoc],
+  lmul_lmul_comm_rmul := λ a b, by rw [←mul_assoc],
+  rmul_comm_rmul_rmul := λ a b, by rw [← mul_assoc, ← mul_assoc], }
+
+@[priority 100] -- see Note [lower instance priority]
+instance comm_semigroup.is_comm_jordan (B : Type u) [comm_semigroup B] : is_comm_jordan B :=
+{ mul_comm := mul_comm,
+  jordan := λ a b, mul_assoc _ _ _, }
+
+variables {A : Type*} [non_unital_non_assoc_ring A]
+
+/-- Left multiplication operator. This is a variant of `add_monoid_hom.mul_left`. -/
+@[simps] def add_monoid.End.L : A →+ add_monoid.End A := add_monoid_hom.mul
+local notation `L` := add_monoid.End.L
+
+/-- Right multiplication operator. This is a variant of `add_monoid_hom.mul_right`. -/
+@[simps] def add_monoid.End.R : A →+ add_monoid.End A := (L : A →+ add_monoid.End A).flip
+local notation `R` := add_monoid.End.R
+
+/-! The Jordan axioms can be expressed in terms of commuting multiplication operators -/
+section lie
+variables [is_jordan A]
+
+@[simp] lemma lie_L_R (a : A) : ⁅L a, R a⁆ = 0 :=
+add_monoid_hom.ext $ λ b, sub_eq_zero_of_eq (is_jordan.lmul_comm_rmul _ _).symm
+
+@[simp] lemma lie_L_L_sq (a : A) : ⁅L a, L (a * a)⁆ = 0 :=
+add_monoid_hom.ext $ λ b, sub_eq_zero_of_eq (is_jordan.lmul_lmul_comm_lmul _ _).symm
+
+@[simp] lemma lie_L_R_sq (a : A) : ⁅L a, R (a * a)⁆ = 0 :=
+add_monoid_hom.ext $ λ b, sub_eq_zero_of_eq (is_jordan.lmul_comm_rmul_rmul _ _).symm
+
+@[simp] lemma lie_L_sq_R (a : A) : ⁅L (a * a), R a⁆ = 0 :=
+add_monoid_hom.ext $ λ b, sub_eq_zero_of_eq (is_jordan.lmul_lmul_comm_rmul _ _)
+
+@[simp] lemma lie_R_R_sq (a : A) : ⁅R a, R (a * a)⁆ = 0 :=
+add_monoid_hom.ext $ λ b, sub_eq_zero_of_eq (is_jordan.rmul_comm_rmul_rmul _ _).symm
+
+end lie
+
+variable [is_comm_jordan A]
+
+/-
+instance : comm_monoid A :=
+{ mul_comm := λ a b, is_comm_jordan.mul_comm a b,
+  .. (show non_unital_non_assoc_ring A, by apply_instance) }
+-/
```

Comment:
Is there a reason this is here? Can we delete it, or put it in a TODO comment if there's still work to be done here?

Reply:
I was asked to add it [here](https://github.com/leanprover-community/mathlib/pull/11073#discussion_r782126142) but I don't think it makes sense. `A` can't be `comm_monoid` as `A` is neither unital or associative. In the other direction we now have `comm_semigroup.is_comm_jordan`. I've deleted it.